sympy/simplify/gammasimp.py

```diff
@@ -0,0 +1,527 @@
+from __future__ import print_function, division
+
+from sympy.core import Function, S, Mul, Pow, Add
+from sympy.core.compatibility import ordered, default_sort_key
+from sympy.core.basic import preorder_traversal
+from sympy.core.function import count_ops, expand_func
+from sympy.functions.combinatorial.factorials import (binomial,
+    CombinatorialFunction, factorial)
+from sympy.functions import gamma, sqrt, sin
+from sympy.polys import factor, cancel
+
+from sympy.utilities.iterables import sift, uniq
+
+
+def gammasimp(expr, as_comb=False):
+    r"""
+    Simplify expressions with gamma functions.
+
+    This function takes as input an expression containing gamma
+    functions or functions that can be rewritten in terms of gamma
+    functions and tries to minimize the number of those functions and
+    reduce the size of their arguments.
+
+    If as_comb is True and every argument of gamma functions is integers,
+    it tries to preserve them and rewrite expression in terms of
+    factorials and binomials after simplification. (False by default)
+
+    The algorithm works by rewriting all gamma functions as expressions
+    involving rising factorials (Pochhammer symbols) and applies
+    recurrence relations and other transformations applicable to rising
+    factorials, to reduce their arguments, possibly letting the resulting
+    rising factorial to cancel. Rising factorials with the second argument
+    being an integer are expanded into polynomial forms and finally all
+    other rising factorial are rewritten in terms of gamma functions.
+
+    Then the following two steps are performed if as_comb is False.
+
+    1. Reduce the number of gammas by applying the reflection theorem
+       gamma(x)*gamma(1-x) == pi/sin(pi*x).
+    2. Reduce the number of gammas by applying the multiplication theorem
+       gamma(x)*gamma(x+1/n)*...*gamma(x+(n-1)/n) == C*gamma(n*x).
+
+    It then reduces the number of prefactors by absorbing them into gammas
+    where possible and expands gammas with rational argument.
+
+    If as_comb is True, it rewrites gammas as factorials and converts
+    (a+b)!/a!b! into binomials.
```

Comment:
I wonder if this flag is really needed here. The former can be done with `rewrite(gamma)` or `replace`, and the latter should be done by combsimp. I get that you want to keep the logic together for now, but perhaps that should be hidden from the user.

Reply:
@asmeurer I agree. How about making _gamma_as_comb a public function with maybe force flag to determine to check for integer arguments?

Reply:
Would it do any more than expr.rewrite(factorial) (or maybe `combsimp(expr.rewrite(factorial))`)?